File: backend/CookingRecipes.API/Models/Recipe.cs
using CookingRecipes.API.Data.Enums;
using CookingRecipes.Data.Enums;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace CookingRecipes.Models
{
    public class Recipe
    {
        public int? Id { get; set; }
        [Required]
        public string Title { get; set; } = string.Empty;
        [Required]
        public string Instructions { get; set; } = string.Empty;
        public string ImageName { get; set;} = string.Empty;
        [Required]
        public string Ingredients { get; set; } = string.Empty;
        public int Difficulty { get; set; } = 0;
        public DishType DishType { get; set; } = 0;
        public CuisineType CuisineType { get; set; } = 0;
        public ICollection<UserFavoriteRecipe>? UserFavoriteRecipes { get; set; }
        public ICollection<Review>? Reviews { get; set; }
        public ICollection<RecipeCategory>? RecipeCategories { get; set; }
    }
}

File: frontend/CookingRecipes.UI/src/app/models/recipe.ts
import { CuisineType } from "./enums/cuisineType"
import { DishType } from "./enums/dishType"
import RecipeCategory from "./recipeCategory"

export default class Recipe {
    id?: number
    title: string
    instructions: string
    imageName: string
    ingredients: string
    difficulty: number
    dishType: DishType
    cuisineType: CuisineType
    recipeCategories: RecipeCategory[]
     
    constructor(
        id: number,
        title: string,
        instructions: string,
        imageName: string,
        ingredients: string,
        difficulty: number,
        dishType: DishType,
        cuisineType: CuisineType,
        recipeCategories: RecipeCategory[]
    ) {
        this.id = id
        this.title = title
        this.instructions = instructions
        this.imageName = imageName
        this.ingredients = ingredients
        this.difficulty = difficulty
        this.dishType = dishType
        this.cuisineType = cuisineType
        this.recipeCategories = recipeCategories
    } 
}